### PR TITLE
fix(date): don't allow parameters <= 0

### PR DESCRIPTION
**Pull request by Shinigami92**
While working on #1534 I found this bug and extracted it to this PR.

Before this fix, `values === 0` would be treated as _falsy_ value and fallback to `1`.
We want to prevent this behavior directly and now throw an error when the user tries to use values below or equals to 0.

### CODE REVIEW

File: src/modules/date/index.ts [hunk 1]
```diff
@@ -47,10 +47,14 @@ export class DateModule {
    * @since 2.0.1
    */
   past(years?: number, refDate?: string | Date | number): Date {
+    if (years <= 0) {
+      throw new FakerError('Years should be greater than 0.');
```
Comment (Shinigami92) [inline src/modules/date/index.ts]: @import-brain should this be `should` or `must`?

Comment (ejcheng) [reply src/modules/date/index.ts]: Hmm, I would say `must` but either would work here.

**Review by ST-DDT — DISMISSED** [review]
As an alternative we could treat the `<=0` as zero returning effectively `now`/`refDate` (minus/plus 1 second to be in the past/future).

File: test/date.spec.ts [hunk 2]
```diff
@@ -158,11 +158,11 @@ describe('date', () => {
...
-        it('should return a future date when years 0', () => {
+        it('should throw an error when years = 0', () => {
           const refDate = new Date();
-          const date = faker.date.future(0, refDate.toISOString());
-
-          expect(date).greaterThan(refDate); // date should be after the date given
+          expect(() => faker.date.future(0, refDate.toISOString())).toThrow(
+            new FakerError('Years must be greater than 0.')
+          );
         });
```
Comment (ST-DDT) [inline test/date.spec.ts:166]: These tests are missing for recent and soon.

Comment (Shinigami92) [reply test/date.spec.ts:166]: Good catch, thx
Will fix this late evening or tomorrow

**Comment by Shinigami92** [comment]
> As an alternative we could treat the `<=0` as zero returning effectively `now`/`refDate` (minus/plus 1 second to be in the past/future).

IMO this is not directly part of this issue. I would like to have this fixed to proceed with #1534.
But yeah, we can talk about this in next meeting in general, but would like to prio it down in the agenda so we have more focus on other more important things already on the agenda.

**Review by ST-DDT — APPROVED** [review]

**Review by ejcheng — APPROVED** [review]

**Review by xDivisionByZerox — APPROVED** [review]

**Pull request merged**